File: DDCharacterCreator/app/src/main/java/com/example/ddcharactercreator/FirstQuestionnaireActivity.java
package com.example.ddcharactercreator;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class FirstQuestionnaireActivity extends AppCompatActivity {



    public FirstQuestionnaireActivity(){
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.first_questionnaire_layout);

        final Spinner raceSpinner = findViewById(R.id.race_spinner);
        final Spinner classSpinner = findViewById(R.id.class_spinner);
        final Spinner alignmentSpinner = findViewById(R.id.alignment_spinner);
        final EditText characterNameEditText = findViewById(R.id.character_name_edit_text);

        final StatNumberPicker strengthStatNumberPicker = findViewById(R.id.strength_level);
        final StatNumberPicker dexterityStatNumberPicker = findViewById(R.id.dexterity_level);
        final StatNumberPicker constitutionStatNumberPicker = findViewById(R.id.constitution_level);
        final StatNumberPicker intelligenceStatNumberPicker = findViewById(R.id.intelligence_level);
        final StatNumberPicker wisdomStatNumberPicker = findViewById(R.id.wisdom_level);
        final StatNumberPicker charismaStatNumberPicker = findViewById(R.id.charisma_level);

        Button nextQuestionnaireButton = findViewById(R.id.next_questionnaire_button);
        nextQuestionnaireButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(characterNameEditText.getText().toString().equals("")){
                    Toast.makeText(getApplicationContext(), "Choose your character's name", Toast.LENGTH_SHORT).show();
                }else {
                    String race = raceSpinner.getSelectedItem().toString();
                    String characterClass = classSpinner.getSelectedItem().toString();
                    String alignment = alignmentSpinner.getSelectedItem().toString();
                    String name = characterNameEditText.getText().toString();

                    ArrayList<Integer> statsValuesList = new ArrayList<Integer>();
                    statsValuesList.add(strengthStatNumberPicker.getValue());
                    statsValuesList.add(dexterityStatNumberPicker.getValue());
                    statsValuesList.add(constitutionStatNumberPicker.getValue());
                    statsValuesList.add(intelligenceStatNumberPicker.getValue());
                    statsValuesList.add(wisdomStatNumberPicker.getValue());
                    statsValuesList.add(charismaStatNumberPicker.getValue());

                    nextQuestionnaire(new Character(1, race, characterClass, alignment, name, statsValuesList));
                }
            }
        });
    }
    private void nextQuestionnaire(Character character){
        Intent intent = new Intent(this, SecondQuestionnaireActivity.class);
        intent.putExtra(SecondQuestionnaireActivity.CHARACTER, character);
        startActivity(intent);
    }
}

File: DDCharacterCreator/app/src/main/java/com/example/ddcharactercreator/FirstQuestionnaireActivity.kt
package com.example.ddcharactercreator

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class FirstQuestionnaireActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.first_questionnaire_layout)
        val raceSpinner = findViewById<Spinner>(R.id.race_spinner)
        val classSpinner = findViewById<Spinner>(R.id.class_spinner)
        val alignmentSpinner = findViewById<Spinner>(R.id.alignment_spinner)
        val characterNameEditText = findViewById<EditText>(R.id.character_name_edit_text)
        val strengthStatNumberPicker = findViewById<StatNumberPicker>(R.id.strength_level)
        val dexterityStatNumberPicker = findViewById<StatNumberPicker>(R.id.dexterity_level)
        val constitutionStatNumberPicker = findViewById<StatNumberPicker>(R.id.constitution_level)
        val intelligenceStatNumberPicker = findViewById<StatNumberPicker>(R.id.intelligence_level)
        val wisdomStatNumberPicker = findViewById<StatNumberPicker>(R.id.wisdom_level)
        val charismaStatNumberPicker = findViewById<StatNumberPicker>(R.id.charisma_level)
        val nextQuestionnaireButton = findViewById<Button>(R.id.next_questionnaire_button)
        nextQuestionnaireButton.setOnClickListener {
            if (characterNameEditText.text.toString() == "") {
                Toast.makeText(applicationContext, "Choose your character's name", Toast.LENGTH_SHORT).show()
            } else {
                val race = raceSpinner.selectedItem.toString()
                val characterClass = classSpinner.selectedItem.toString()
                val alignment = alignmentSpinner.selectedItem.toString()
                val name = characterNameEditText.text.toString()
                val statsValuesList = ArrayList<Int>()
                statsValuesList.add(strengthStatNumberPicker.value)
                statsValuesList.add(dexterityStatNumberPicker.value)
                statsValuesList.add(constitutionStatNumberPicker.value)
                statsValuesList.add(intelligenceStatNumberPicker.value)
                statsValuesList.add(wisdomStatNumberPicker.value)
                statsValuesList.add(charismaStatNumberPicker.value)
                nextQuestionnaire(Character(1, race, characterClass, alignment, name, statsValuesList))
            }
        }
    }

    private fun nextQuestionnaire(character: Character) {
        val intent = Intent(this, SecondQuestionnaireActivity::class.java)
        intent.putExtra(SecondQuestionnaireActivity.CHARACTER, character)
        startActivity(intent)
    }
}